modules/python/init.zsh

```diff
@@ -11,12 +11,12 @@
 if [[ -s "$HOME/.pyenv/bin/pyenv" ]]; then
   path=("$HOME/.pyenv/bin" $path)
   export PYENV_ROOT=$(pyenv root)
-  eval "$(pyenv init -)"
+  eval "$(pyenv init - zsh --no-rehash)"
 
 # Load package manager installed pyenv into the shell session.
 elif (( $+commands[pyenv] )); then
   export PYENV_ROOT=$(pyenv root)
-  eval "$(pyenv init -)"
+  eval "$(pyenv init - zsh --no-rehash)"
```

Comment:
Using `zsh` here already makes sense by itself - and could go into a separate PR / commit.